apps/firehose_to_es_processor/test/test_es_client.py

```diff
@@ -0,0 +1,46 @@
+from lib.es_client import ESClient
+from lib.firehose_record_processor import FirehoseRecordProcessor
+import unittest
+import datetime
+import os
+import time
+
+
+class TestESClient(unittest.TestCase):
+    test_prefix_ele = '.'.join(
+        [e for e in [os.environ.get('TRAVIS_BUILD_ID'), os.environ.get('TRAVIS_EVENT_TYPE')] if e]
+    )
+    index_prefix = f"test{('.' + test_prefix_ele) if test_prefix_ele else ''}"""
+    es_client = ESClient()
+    es = es_client.es
+
+    def test_create_cwl_day_index(self):
+        try:
+            index_name = self.es_client._format_today_index_name(self.index_prefix)
+            if self.es.indices.exists(index_name):
+                self.es_client.delete_index(index_name)
+            self.assertEqual(self.es.indices.exists(index_name), False)
+            self.es_client.create_cwl_day_index(self.index_prefix)
+            self.assertEqual(self.es.indices.exists(index_name), True)
+        finally:
+            if self.es.indices.exists(index_name):
+                self.es_client.delete_index(index_name)
+
+    def test_bulk_post(self):
+        try:
+            index_name = self.es_client._format_today_index_name(self.index_prefix)
+            if not self.es.indices.exists(index_name):
+                self.es_client.create_cwl_day_index(self.index_prefix)
+            self.assertEqual(self.es.indices.exists(index_name), True)
+            data = [{"owner": "test_owner", "logGroup": "/test/test_log_group", "logStream": "test_log_stream", "messageType": 'DATA_MESSAGE'}]
+            data[0]["logEvents"] = [{"id": 123456, "timestamp": 1519970297000, "message": 'with_json{"hi": "hello"}with_json'}, {"id": 123456, "timestamp": 1519970297000, "message": 'with_json{"hi": "hello"}with_json'}]
+            firehose_record_processor = FirehoseRecordProcessor(data)
+            firehose_record_processor.run()
+            output_records = firehose_record_processor.output_records
+            self.es_client.bulk_post(output_records, self.index_prefix)
+            time.sleep(5)
```

Comment:
I would do this with an `@eventually`-like function instead. This should make the test faster/more robust. See https://github.com/HumanCellAtlas/data-store/blob/557e60e3672075d532d790bca4422ae74023ea0a/tests/__init__.py#L45